update/process/shardBlock.go

```diff
@@ -171,6 +190,135 @@ func (s *shardBlockCreator) createMiniBlockHeaders(body *block.Body) (int, []blo
 	return totalTxCount, miniBlockHeaders, nil
 }
 
+func (s *shardBlockCreator) saveAllTransactionsToStorageIfSelfShard(
+	shardHdr *block.Header,
+	body *block.Body,
+) {
+	if shardHdr.GetShardID() != s.selfShardID {
+		return
+	}
+
+	mapBlockTypesTxs := make(map[block.Type]map[string]data.TransactionHandler)
+	for i := 0; i < len(body.MiniBlocks); i++ {
+		miniBlock := body.MiniBlocks[i]
+		if _, ok := mapBlockTypesTxs[miniBlock.Type]; !ok {
+			mapBlockTypesTxs[miniBlock.Type] = s.txCoordinator.GetAllCurrentUsedTxs(miniBlock.Type)
+		}
+
+		marshalizedMiniBlock, errNotCritical := s.marshalizer.Marshal(miniBlock)
+		if errNotCritical != nil {
+			log.Warn("saveAllTransactionsToStorageIfSelfShard.Marshal", "error", errNotCritical.Error())
+			continue
+		}
+
+		errNotCritical = s.storage.Put(dataRetriever.MiniBlockUnit, shardHdr.MiniBlockHeaders[i].Hash, marshalizedMiniBlock)
+		if errNotCritical != nil {
+			log.Warn("saveAllTransactionsToStorageIfSelfShard.Put -> MiniBlockUnit", "error", errNotCritical.Error())
+		}
+	}
+
+	mapTxs := s.importHandler.GetTransactions()
+	// save transactions from imported map
+	for _, miniBlock := range body.MiniBlocks {
+		if miniBlock.SenderShardID == s.selfShardID {
+			continue
+		}
+
+		for _, txHash := range miniBlock.TxHashes {
+			tx, ok := mapTxs[string(txHash)]
+			if !ok {
+				log.Warn("miniblock destination me in genesis block should contain only imported txs")
+				continue
+			}
+
+			unitType := dataRetriever.TransactionUnit
```

Comment:
You can extract code 234-242 in one method getUnitType for example, as this method is too big

Reply:
done